docs/introduction/migration-v4-v5.md

```diff
@@ -0,0 +1,422 @@
+
+### New Channel Attribute Interfaces
+
+`GuildChannel` now implements `getGuild` and `getManager` *only*. It's meant to serve as a generic type to hold channels from guilds.
+
+Specific channel attributes, such as slowmode and permissions, have been split into several interfaces. These interfaces can be found in [the `net.dv8tion.jda.api.entities.channel.attribute` package](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/entities/channel/attribute/package-summary.html). Each interface in this package extends `GuildChannel`.
+
+### Permission Access Changes
+
+With JDA v5 introducing a more butchered down `GuildChannel` interface, `GuildChannel#getPermissionContainer() : IPermissionContainer` has been introduced to make accessing permissions easier. This allows for users to confidently access the entity that dictates the permissions for a given channel without having to check whether the channel actually supports permissions, such as in the case of threads.
+
+Of course, if you have a channel type that already supports `IPermissionContainer`, you don't need this getter.
+
+### Type-Trimmed Channel Managers
+
+In JDA v4, `GuildChannel#getManager` returned a `ChannelManager` that gave every possible setter for every channel type. This caused for `UnsupportedOperationException` or `IllegalStateException` to be thrown in some cases, such as calling `setBitrate` on a `TextChannel`.
+
+JDA v5 provides type-trimmed channel managers, which provide only the setters that we know *for sure* can work on the given channel. Each of these managers can be found in [the `net.dv8tion.jda.api.managers.channel` package](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/managers/channel/package-summary.html). These all follow the same implement/extension hierarchy as the channels do and map 1:1.
+
+### Independant Stage Channel and News Channel Entities
+
+[`StageChannel`](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/entities/channel/concrete/StageChannel.html) and [`NewsChannel`](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/entities/channel/concrete/NewsChannel.html) were previously variants of `VoiceChannel` and `TextChannel`. These are now their own independant entities. This comes with a number of changes:
+
+- `ChannelAction#setNews` has been removed in favor of `ChannelAction.setType`
+- `ChannelManager#getType` was removed in favor of `ChannelManager#getChannel()#getType`
+- `ChannelManager#setNews` has been removed in favor of `ChannelManager#setType`
+- `TextChannel#crosspostMessage` was removed in favor of `NewsChannel#crosspostMessage`
+- `TextChannel#follow` has been removed in favor of `NewsChannel#follow`
+- `TextChannel#isNews` has been removed
+
```

Comment:
Having both `has been removed in favor of` and `was removed in favor of` seems inconsistent